autoscale/src/main/java/com/sequenceiq/periscope/repository/ClusterRepository.java

```diff
@@ -71,6 +71,12 @@ List<Long> findClusterIdsByStackTypeAndPeriscopeNodeId(@Param("stackType") Stack
     List<Long> findClusterIdsByStackTypeAndPeriscopeNodeIdAndAutoscalingEnabled(@Param("stackType") StackType stackType,
             @Param("periscopeNodeId") String periscopeNodeId, @Param("autoscalingEnabled") Boolean autoscalingEnabled);
 
+    @Query("SELECT distinct c.id FROM Cluster c JOIN c.scalingTriggers sc WHERE c.stackType = :stackType AND c.autoscalingEnabled = :autoscalingEnabled " +
```

Comment:
A little confused by what this query does ... specifcally JOIN c.scalingTriggers AND sc is NOT NULL.

Another point of consideration - what happens when autoscaling is disabled for a cluster for which we have not yet updated the status of the trigger (or performed the relevant cleanup).

Reply:
Removed this